libs/workspace/src/schematics/dt-component-example/index.ts

```diff
@@ -0,0 +1,258 @@
+  ).pop() as ts.ExportDeclaration;
+  const end = lastExport.end + 1;
+  const toInsertExport = `export * from './${exportPath}';`;
+  return new InsertChange(modulePath, end, toInsertExport);
+}
+
+function updateModules(options: DtExampleExtendedOptions): Rule {
+  return (host: Tree) => {
+    // add import
+    const modulePath = join(
+      'libs',
+      'examples',
+      'src',
+      strings.dasherize(options.component),
+      `${strings.dasherize(options.component)}-examples.module.ts`,
+    );
+    const sourceFile = getSourceFile(host, modulePath);
+
+    const importChange = addImport(
+      sourceFile,
+      options.exampleComponent.component,
+      options.exampleRoute,
+      modulePath,
+    );
+
+    // find last module in the array, and add new module
+    const modulesDeclaration = findNodes(
+      sourceFile,
+      ts.SyntaxKind.VariableDeclaration,
+    ).find(
+      (node: ts.VariableDeclaration) =>
+        node.name.getText() === options.exampleComponent.modulesConstant,
+    ) as ts.VariableDeclaration;
+    const modulesElements = (modulesDeclaration.initializer as ts.ArrayLiteralExpression)
+      .elements;
+    const lastElement = modulesElements[modulesElements.length - 1];
+    const end = modulesElements.hasTrailingComma
+      ? lastElement.getEnd() + 1
+      : lastElement.getEnd();
+    const routesChange = new InsertChange(
+      modulePath,
+      end,
+      `${modulesElements.hasTrailingComma ? ' ' : ', '}${
+        options.exampleComponent.component
+      }`,
+    );
+
+    return commitChanges(host, [importChange, routesChange], modulePath);
+  };
+}
+
+function updateIndex(options: DtExampleExtendedOptions): Rule {
+  return (host: Tree) => {
+    // add export to index.ts
+    const indexPath = join(
+      'libs',
+      'examples',
+      'src',
+      strings.dasherize(options.component),
+      'index.ts',
+    );
+    const sourceIndexFile = getSourceFile(host, indexPath);
+    const exportChange = addExport(
+      sourceIndexFile,
+      options.exampleRoute,
+      indexPath,
+    );
+
+    return commitChanges(host, exportChange, indexPath);
+  };
+}
+
+export default function (options: DtComponentExampleOptions): Rule {
+  return async (tree: Tree) => {
+    const exampleId = `${strings.dasherize(
+      options.component,
+    )}-${strings.dasherize(options.name)}`;
+    const extendedOptions: DtExampleExtendedOptions = {
+      ...options,
+      componentSelector: `dt-${strings.dasherize(options.component)}`,
+      selector: `dt-example-${strings.dasherize(
+        options.name,
+      )}-${strings.dasherize(options.component)}`,
+      componentModule: generateComponentOptions(options),
+      exampleComponent: generateExampleComponentOptions(options),
+      examplesModule: `DtExamples${strings.classify(options.component)}Module`,
+      exampleRoute: `${exampleId}-example/${exampleId}-example`,
+    };
+
+    const templateSource = apply(url('./files'), [
```

Comment:
I would not nest the example folder I would leverage the move function that is provided from angular-devkit.

so keep a flat structure in the files folder `./files/example.component.html`, `./files/example.component.scss` and after the template function add the move function to the destination  `move(normalize('libs/your/path'))`

It is the recommended option by angular as well as it makes refactoring paths way easier as you can search and replace paths via the search functionality from VS code
https://angular.io/guide/schematics-for-libraries#define-the-rule

Reply:
I thought that we discussed otherwise, but maybe I misunderstood, will be fixed :).